app/javascript/components/shared_components/Header.jsx

```diff
@@ -45,7 +45,7 @@ export default function Header() {
             }
             <NavDropdown.Item as={Link} to="/">Need help?</NavDropdown.Item>
             <NavDropdown.Divider />
-            <NavDropdown.Item as={Link} to="/" onClick={deleteSession.mutate}>Sign Out</NavDropdown.Item>
```

Comment:
This had to change due to a bug when singed in user would sign out.
The link would redirect to the `HomePage` before the mutation for signing out the user was even done, this was problematic since it would go into the `useEffect` in `HomePage` and still think that the user is signed in, then redirecting them to the wrong page.

We instead are going to redirect in the mutation and await the query to finish before redirecting. This way the `currentUser` in `HomePage` will have the right information